app/models/concerns/fog_extensions/vsphere/server.rb

```diff
@@ -35,7 +35,10 @@ def scsi_controller_type
       end
 
       def select_nic(fog_nics, nic)
-        fog_nics.detect {|fn| fn.network == nic.compute_attributes['network']} # grab any nic on the same network
+        nic_attrs = nic.compute_attributes
+        match =   fog_nics.detect { |fn| fn.network == nic_attrs['network'] } # grab any nic on the same network
+        match ||= fog_nics.detect { |fn| fn.network  == service.get_network(nic_attrs['network'], datacenter)[:id]  } # no network? try the portgroup
```

Comment:
fn.network appears to be a string according to the ticket, so this looks reasonable.  Testing should confirm..